### PR TITLE
Rework API errors

### PR DESCRIPTION
**Pull request by ThibautBremand**

### CODE REVIEW

File: packages/api/src/getAddress/getAddress.ts [hunk 1]
```diff
@@ -20,8 +30,16 @@ export const getAddress = async (): Promise<GetAddressResponse> => {
...
     };
-    const { result } = await sendMessageToContentScript(message);
-    response.result = result;
+    const { result, error } = await sendMessageToContentScript(message);
+    const parsedError = error ? deserializeError(error) : undefined;
+    if (parsedError) {
+      throw parsedError;
+    }
+
+    if (result) {
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/api/src/getAddress/getAddress.ts:39]: When both the result and the error are undefined, we consider that the user has declined the tx

File: packages/extension/src/chromeServices/content.ts [hunk 2]
```diff
@@ -359,12 +360,13 @@ setTimeout(() => {
...
                 if (type === 'RECEIVE_SEND_PAYMENT/V3') {
-                  const { result } = payload;
+                  const { result, error } = payload;
                   window.postMessage(
                     {
                       source: 'GEM_WALLET_MSG_RESPONSE',
                       messagedId,
-                      result
+                      result,
+                      error
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/extension/src/chromeServices/content.ts:377]: Done

File: packages/api/src/getNetwork/getNetwork.ts [hunk 3]
```diff
@@ -4,9 +4,9 @@ import {
   RequestGetNetworkMessage,
   ResponseType
 } from '@gemwallet/constants';
-import { deserializeError } from '@gemwallet/extension/src/utils/errors';
 
 import { sendMessageToContentScript } from '../helpers/extensionMessaging';
+import { deserializeError } from '../utils/errors';
```
Comment (FlorianBouron) [inline packages/api/src/getNetwork/getNetwork.ts:9]: Seems that this could actually go to the `helpers` folder

Comment (ThibautBremand) [reply packages/api/src/getNetwork/getNetwork.ts]: Done

File: packages/api/src/sendPayment/sendPayment.ts [hunk 4]
```diff
@@ -1,27 +1,43 @@
...
+  ResponseType,
+  SendPaymentRequest,
   SendPaymentResponse
 } from '@gemwallet/constants';
 
 import { sendMessageToContentScript } from '../helpers/extensionMessaging';
+import { deserializeError } from '../utils/errors';
 
 export const sendPayment = async (paymentPayload: SendPaymentRequest) => {
   /* string: hash of the transaction
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/api/src/sendPayment/sendPayment.ts]: Done

File: packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx [hunk 5]
```diff
@@ -79,7 +80,8 @@ export const ConfirmPayment: FC<ConfirmPaymentProps> = ({
         setTransaction(TransactionStatus.Success);
       })
       .catch((e) => {
-        setErrorRequestRejection(e.message);
+        const UIError = toUIError(e);
+        setErrorRequestRejection(UIError.message);
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx:84]: No need to create a constant, it's used only one time:
```suggestion
        setErrorRequestRejection(toUIError(e).message);
```

File: packages/extension/src/components/pages/ShareNFT/ShareNFT.tsx [hunk 6]
```diff
@@ -85,20 +95,20 @@ export const ShareNFT: FC = () => {
   );
 
   const handleReject = useCallback(() => {
-    handleSendMessage(null);
+    handleSendMessage({});
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/ShareNFT/ShareNFT.tsx:98]: Is this working?
Because as per the code it seems that handleSendMessage will be returning everytime: `type: ResponseType.Response,`

File: packages/extension/src/contexts/LedgerContext/LedgerContext.tsx [hunk 7]
```diff
@@ -140,56 +140,29 @@ const LedgerProvider: FC = ({ children }) => {
         throw new Error('You need to have a wallet connected to make a transaction');
       } else {
         // Prepare the transaction
-        try {
```
Comment (FlorianBouron) [inline packages/extension/src/contexts/LedgerContext/LedgerContext.tsx:143]: Why did you remove the try?
